libs/vm/src/vm-imports.ts

```diff
@@ -113,6 +115,49 @@ export default class VmImports {
 		}
 	}
 
+	secp256k1Verify(
+		messagePtr: number,
+		messageLength: number,
+		signaturePtr: number,
+		signatureLength: number,
+		publicKeyPtr: number,
+		publicKeyLength: number,
+	) {
+		const message = Buffer.from(
+			new Uint8Array(
+				this.memory?.buffer.slice(messagePtr, messagePtr + messageLength) ?? [],
+			),
+		);
+		const signature = Buffer.from(
+			new Uint8Array(
+				this.memory?.buffer.slice(
+					signaturePtr,
+					signaturePtr + signatureLength,
+				) ?? [],
+			),
+		);
+		const publicKey = Buffer.from(
+			new Uint8Array(
+				this.memory?.buffer.slice(
+					publicKeyPtr,
+					publicKeyPtr + publicKeyLength,
+				) ?? [],
+			),
+		);
+
+		try {
+			this.callResult = secp256k1Verify(message, signature, publicKey);
```

Comment:
I believe true-myth is in this project. It would be better to do:
```
const result = trySync(() => secp...)
```

instead of doing try/catch